extension/sigv4authextension/extension.go

```diff
@@ -0,0 +1,197 @@
+	"go.opentelemetry.io/collector/config/configauth"
+	"go.uber.org/zap"
+	grpcCredentials "google.golang.org/grpc/credentials"
+)
+
+// Sigv4Auth is a struct that implements the configauth.ClientAuthenticator interface. 
+// It provides the implementation for providing Sigv4 authentication for HTTP requests only.
+type Sigv4Auth struct {
+	cfg 				*Config
+	logger 				*zap.Logger
+	awsSDKInfo 			string
+	componenthelper.StartFunc	// embedded default behavior to do nothing with Start()
+	componenthelper.ShutdownFunc	// embedded default behavior to do nothing with Shutdown()
+}
+
+// compile time check that the Sigv4Auth struct satisfies the configauth.ClientAuthenticator interface
+var _ configauth.ClientAuthenticator = (*Sigv4Auth)(nil)
+
+// SigningRoundTripper is a custom RoundTripper that performs AWS Sigv4.
+type SigningRoundTripper struct {
+	transport 	http.RoundTripper
+	signer 		*sigv4.Signer
+	region 		string
+	service 	string
+	awsSDKInfo 	string
+}
+
+// RoundTripper() returns a custom SigningRoundTripper.
+func (sa *Sigv4Auth) RoundTripper(base http.RoundTripper) (http.RoundTripper, error) {
+	cfg := sa.cfg
+
+	signer := sigv4.NewSigner(cfg.creds)
+
+	// Create the SigningRoundTripper struct
+	rt := SigningRoundTripper {
+		transport: base,
+		signer: signer,
+		region: cfg.Region,
+		service: cfg.Service,
+		awsSDKInfo: sa.awsSDKInfo,
+	}
+	   
+	return &rt, nil
+}
+
+// PerRPCCredentials() is implemented to satisfy the configauth.ClientAuthenticator
+// interface but will not be implemented.
+func (sa *Sigv4Auth) PerRPCCredentials() (grpcCredentials.PerRPCCredentials, error) {
+	return nil, errors.New("Not Implemented")
+}
+
+// RoundTrip() executes a single HTTP transaction and returns an HTTP response, signing
+// the request with Sigv4.
+func (si *SigningRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	reqBody, err := req.GetBody()
+	if err != nil {
+		return nil, err
+	}
+    
+	content, err := ioutil.ReadAll(reqBody)
+	reqBody.Close()
+	if err != nil {
+	    	return nil, err
+	}
+	body := bytes.NewReader(content)
+    
+	// Clone request to ensure thread safety.
+	req2 := cloneRequest(req)
+    
+	// Add the runtime information to the User-Agent header of the request
+	ua := req2.Header.Get("User-Agent")
+	if len(ua) > 0 {
+		ua = ua + " " + si.awsSDKInfo
+	} else {
+		ua = si.awsSDKInfo
+	}
+	req2.Header.Set("User-Agent", ua)
+    
+	// Sign the request
+	_, err = si.signer.Sign(req2, body, si.service, si.region, time.Now())
+	if err != nil {
+		return nil, fmt.Errorf("error signing the request: %v", err)
+	}
+    
+	// Send the request 
+	resp, err := si.transport.RoundTrip(req2)
+	if err != nil {
+		return nil, err
+	}
+       
+       
+       return resp, nil
+} 
+
+// newSigv4Extension() is called by createExtension() in factory.go and 
+// returns a new Sigv4Auth struct.
+func newSigv4Extension(cfg *Config, awsSDKInfo string, logger *zap.Logger) (*Sigv4Auth, error) {
+	err := cfg.Validate()
```

Comment:
I don't believe this is necessary.  The config would have been validated when the associated pipeline was constructed and not changed since.

Reply:
Removed and made changes to tests that tested this validation